bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -218,6 +218,62 @@ impl Script {
         }
     }
 
+    /// Checks whether a script pubkey is a bare multisig output.
+    #[inline]
+    pub fn is_multisig(&self) -> bool {
+        if self.0.is_empty() {
+            return false;
+        }
+
+        let required_sigs;
+
+        let mut instructions = self.instructions();
+        if let Some(Ok(Instruction::Op(op))) = instructions.next() {
+            if let Some(pushnum) = op.decode_pushnum() {
+                required_sigs = pushnum;
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+
+        let mut num_pubkeys: u8 = 0;
+        while let Some(Ok(instruction)) = instructions.next() {
+            match instruction {
+                Instruction::PushBytes(_) => {
+                    num_pubkeys += 1;
+                }
+                Instruction::Op(op) => {
+                    if let Some(pushnum) = op.decode_pushnum() {
+                        if pushnum != num_pubkeys {
+                            return false;
+                        }
+                    }
+                    break;
+                }
+            }
+        }
+
+        if required_sigs > num_pubkeys {
+            return false;
+        }
+
+        if let Some(Ok(Instruction::Op(op))) = instructions.next() {
+            if op != OP_CHECKMULTISIG {
+                return false;
+            }
+        } else {
+            return false;
+        }
+
+        if !instructions.next().is_none() {
+            return false;
+        }
+
+        true
```

Comment:
Perhaps just return the result of `next().is_none()`
```
instructions.next().is_none()
```

Reply:
Updated in de5103b9d6e3c2acd970da6e5b6036ca16c690cb